mono/mini/debugger-agent.c

```diff
@@ -3649,7 +3773,7 @@ process_event (EventKind event, gpointer arg, gint32 il_offset, MonoContext *ctx
 			thread = NULL;
 		} else {
 			if (!thread)
-				thread = is_debugger_thread () ? mono_thread_get_main () : mono_thread_current ();
+				thread = mono_thread_current ();
```

Comment:
Feels fishy

Reply:
Indeed, I think this is incorrect. I've reverted this change.